### PR TITLE
Elasticsearch: Add frozen indices search support

### LINKED ISSUE #18277: Add query param ignore_throttled to Elasticsearch datasource
**What would you like to be added**:
Add query param ignore_throttled to Elasticsearch datasource
 https://www.elastic.co/guide/en/elasticsearch/reference/current/searching_a_frozen_index.html

**Why is this needed**:
This param is needed then in newer Elasticsearch version frozen indices is used. This is especially useful with time series data.
https://www.elastic.co/guide/en/elasticsearch/reference/current/frozen-indices.html

### PR DESCRIPTION
**Pull request by Elfo404**
Re-picking https://github.com/grafana/grafana/pull/27472

What this PR does / why we need it:
This PR adds a configuration option for elasticsearch 6.6.0+ to allow search requests to be executed against frozen indices.

The option is off by default to not change previous behavior.

See:
https://www.elastic.co/guide/en/elasticsearch/reference/current/frozen-indices.html
https://www.elastic.co/guide/en/elasticsearch/reference/current/searching_a_frozen_index.html

Which issue(s) this PR fixes:

Fixes #18277

**HOW TO TEST**

Using `elastic77` docker block, freeze an index and configure your data source to use the same index.

When the index is frozen and the `Include Frozen Indices` is not active you should get no data (also in alerting), when is active then everything should be normal.

Querying frozen indices is only available in X-Pack with ES >= 6.6.0, so be sure the x-pack flag is set to true in datasource settings.
Screenshot 2021-06-22 at 10 22 10

to freeze an index: `curl -X POST "localhost:13200/metrics-2021.06.22/_freeze?pretty` (replace `metrics-2021.06.22` with your index name)

/cc. @estermv @simianhacker

### CODE REVIEW

File: docs/sources/datasources/elasticsearch.md [hunk 1]
```diff
@@ -80,6 +80,13 @@ number followed by a valid time identifier, e.g. `1m` (1 minute) or `30s` (30 se
 | `s`        | second      |
 | `ms`       | millisecond |
 
+### X-Pack Enabled
```
Comment (oddlittlebird) [inline docs/sources/datasources/elasticsearch.md:83]: Why are the two headings here in title case? If you are not matching the UI, then headings should be in sentence case.

Comment (Elfo404) [reply docs/sources/datasources/elasticsearch.md:83]: It is matching the UI, but probably we should change the UI to be sentence case?

Comment (achatterjee-grafana) [reply docs/sources/datasources/elasticsearch.md:83]: @Elfo404 - yes, please! TY

Comment (Elfo404) [reply docs/sources/datasources/elasticsearch.md]: @achatterjee-grafana @oddlittlebird done :) thanks for the feedback :)

**Comment by gabor** [comment]
there is a complication when ngalert, with a frozen index, and use-frozen-disabled. running the alert-query produces this warning: "Error: Failed to evaluate queries and expressions: failed to execute conditions: alert rule panic; please check the logs for the full stack”

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 2]
```diff
@@ -778,11 +780,17 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
...
-      return `_msearch?max_concurrent_shard_requests=${this.maxConcurrentShardRequests}`;
+      searchParams.append('max_concurrent_shard_requests', '' + this.maxConcurrentShardRequests);
+    }
+
+    if (gte(this.esVersion, '6.6.0') && this.xpack && this.includeFrozen) {
+      searchParams.append('ignore_throttled', 'false');
     }
 
-    return '_msearch';
+    return (`_msearch?` + searchParams.toString()).replace(/\?+$/, '');
```
Comment (gabor) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:793]: is the `+` in the regex necessary? my understanding is that you want to remove the `?` from the end when it is at the end (meaning no query-params), so `/\?$/` should be enough, but maybe i'm missing something?

Comment (gabor) [reply public/app/plugins/datasource/elasticsearch/datasource.ts:793]: i usually do not use backticks-around-strings when i do not need to add variables to the string, so i recommend making these normal single-quotes, but this is just nitpicking, feel free to ignore it 😄

**Review by achatterjee-grafana — APPROVED** [review]
Doc content looks good.

File: docs/sources/datasources/elasticsearch.md [hunk 3]
```diff
@@ -80,6 +80,13 @@ number followed by a valid time identifier, e.g. `1m` (1 minute) or `30s` (30 se
 | `s`        | second      |
 | `ms`       | millisecond |
 
+### X-Pack enabled
+
+Enables `X-Pack` specific features and options, providing the query editor with additional aggregations such as `Rate` and `Top Metrics`.
+
+#### Include frozen indices
+
+When `X-Pack enabled` is active and the configured Elasticsearch version is `>=6.6.0`, you can configure Grafana to not ignore [frozen indices](https://www.elastic.co/guide/en/elasticsearch/reference/7.13/frozen-indices.html) when performing search requests.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/elasticsearch.md:89]: "Elasticsearch version is higher than 6.6.0" is what I will recommend. But not a blocker.

File: public/app/plugins/datasource/elasticsearch/datasource.ts (hunk 2 above)
Comment (Elfo404) [reply public/app/plugins/datasource/elasticsearch/datasource.ts]: you are very much right on both points

File: docs/sources/datasources/elasticsearch.md (hunk 3 above)
Comment (Elfo404) [reply docs/sources/datasources/elasticsearch.md]: done! 😺

**Review by gabor — APPROVED** [review]
LGTM! it works fine with the alerts too!

**Review by dsotirakis — APPROVED** [review]
LGTM!

**Pull request merged**